include/nbl/video/IGPUCommandPool.h

```diff
@@ -13,7 +14,38 @@ namespace nbl::video
 
 class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
 {
+public:
+        static inline constexpr uint32_t COMMAND_SEGMENT_SIZE = 128u << 10u;
+
+private:
+        static inline constexpr uint32_t COMMAND_ALIGNMENT = 64u;
+
+        static inline constexpr uint32_t COMMAND_SEGMENT_ALIGNMENT = 64u;
+
+        static inline constexpr uint32_t MAX_COMMAND_SEGMENT_BLOCK_COUNT = 3u;// 16u;
+        static inline constexpr uint32_t COMMAND_SEGMENTS_PER_BLOCK = 2u; // 256u;
+        static inline constexpr uint32_t MIN_POOL_ALLOC_SIZE = COMMAND_SEGMENT_SIZE;
+
     public:
+        class ICommand;
+        class CommandSegment;
+
+        class CBindIndexBufferCmd;
+
+        class CDrawCmd;
+        class CDrawIndexedCmd;
+
+        class CDrawIndirectCommonBase;
+        class CDrawIndirectCmd;
+        class CDrawIndexedIndirectCmd;
+        class CDrawIndirectCountCmd;
+        class CDrawIndexedIndirectCountCmd;
+
+        class CBeginRenderPassCmd;
+        class CEndRenderPassCmd;
```

Comment:
these should probably be protected.

Reply:
Then make `IGPUCommandBuffer` a `friend`?